src/compositions/usePrefetchedProps.ts

```diff
@@ -0,0 +1,30 @@
+import { MaybeRef, Ref, ref, watch } from 'vue'
+import { usePropStore } from '@/compositions/usePropStore'
+import { PrefetchConfigs } from '@/types/prefetch'
+import { ResolvedRoute } from '@/types/resolved'
+
+export type UsePrefetchedProps = {
+  prefetchProps: Ref<Record<string, unknown>>,
+  commitPrefetchedProps: () => void,
+}
+
+export function usePrefetchedProps(route: MaybeRef<ResolvedRoute | undefined>, prefetch: MaybeRef<PrefetchConfigs | undefined>): UsePrefetchedProps {
+  const routeRef = ref(route)
+  const prefetchRef = ref(prefetch)
+  const store = usePropStore()
+  const prefetchProps = ref<Record<string, unknown>>({})
```

Comment:
this isn't reactive so we should just use a `let` here.  Also, we're in a prefetch props composition. We could probably simplify some of these names. 
```suggestion
  const props: Record<string, unknown> = {}
```

Reply:
I see you are exposing it. I guess I wouldn't expose it and I wouldn't make it reactive. That would be my suggestion.